compiler/circle-mpqsolver/src/bisection/VISQErrorApproximator.test.cpp

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2023 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "VISQErrorApproximator.h"
+
+#include <json.h>
+#include <fstream>
+#include <unistd.h>
+#include <gtest/gtest.h>
+
+namespace
+{
+
+void writeDataToFile(const std::string &path, const std::string &data)
+{
+  std::ofstream file;
+  file.open(path);
+  file << data;
+  file.close();
+}
+
+std::string makeTemporaryFile()
+{
+  FILE *file = std::tmpfile();
+  char fname[FILENAME_MAX], link[FILENAME_MAX] = {0};
+  sprintf(fname, "/proc/self/fd/%d", fileno(file));
```

Comment:
sorry for recommend `tmpfile()`
I couldn't catch your intention of temporary file.
Maybe https://en.cppreference.com/w/cpp/io/c/tmpnam would be better usage for this.
Can you please revise this once more?

Why?
- using system `/tmp/~~~` string looks system-ish
- `/proc/self/fd` also looks system-ish

Reply:
Just FYI
https://stackoverflow.com/questions/21295974/are-there-any-reasons-to-avoid-using-tmpnam-to-get-a-name-for-a-temporary-file,
IMHO seems like `tempnam` is not recommended for usage.
also from https://man7.org/linux/man-pages/man3/tmpnam.3.html
```
Note: avoid using these functions; use [mkstemp(3)](https://man7.org/linux/man-pages/man3/mkstemp.3.html) or [tmpfile(3)](https://man7.org/linux/man-pages/man3/tmpfile.3.html)
       instead.
```
May be return to `mkstemp`?

Reply:
> May be return to mkstemp?

it requires system-ish fixed path to temporary filename. I want to avoid using this.

Reply:
ok

Reply:
@seanshpark 
Sorry for use of '/tmp` .
For `mkstemp` `tmp` may be removed , so that the name can be resolved to 'VISQERRORApproximator-TEST-XXXXXX`.
`tmp` was added for clarity that it's temporary file.